### DISCUSSION TITLE
Can I control which SCIP heuristic to run via ecole?

### DISCUSSION THREAD

**Discussion by rybindmitry** [ds4dm/ecole#328] [Questions]
Let's say I want to collect data about SCIP heuristics efficiency in different B&B nodes. Is this possible with ecole? I couldn't find proper commands in any of the environments.

**Comment by AntoinePrv** [comment]
Hi @rybindmitry,

I am not sure exactly what you are trying to achieve here. Do you have an SCIP/PySCIPOpt equivalent?
Maybe this would help.
- We do not have "SCIP heuristic" environment where you could train an agent to select among existing heuristics at each BnB node (that would be nice to have).
- If you want to force activation/deactivation of certain SCIP heuristics, you can use the SCIP parameters to your environment constructor.
- If you want to collect statistics about the heuristic, you can use PySCIPOpt callbacks with Ecole. Use `env.model.as_pyscipopt()` and add as usual.
